components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -1059,6 +1174,68 @@ public static String getFileContentAsJson(String pathToArchive) throws IOExcepti
         return jsonContent;
     }
 
+    /**
+     * Retrieve Policy Definition as JSON content.
+     *
+     * @param pathToArchive Path to Policy archive
+     * @throws IOException If an error occurs while reading the file
+     */
+    public static String getPolicyFileContentAsJson(String pathToArchive) throws IOException {
+
+        String jsonContent = null;
+        String pathToYamlFile = pathToArchive + ImportExportConstants.YAML_EXTENSION;
+        String pathToJsonFile = pathToArchive + ImportExportConstants.JSON_EXTENSION;
+
+        // Load yaml representation first if it is present
+        if (CommonUtil.checkFileExistence(pathToYamlFile)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Found policy definition file " + pathToYamlFile);
+            }
+            String yamlContent = FileUtils.readFileToString(new File(pathToYamlFile));
+
+            jsonContent = CommonUtil.yamlToJson(yamlContent);
+            JsonObject object = new JsonParser().parse(jsonContent).getAsJsonObject();
+            JsonElement data = object.get("data");
+            object.remove("data");
+
+            for (String key : data.getAsJsonObject().keySet()) {
+                object.add(key, data.getAsJsonObject().get(key));
+            }
+
+            jsonContent = object.toString();
```

Comment:
Can't we reuse the logic that we used for APIs here as well? Refactor that logic to some function and use the same function in every place?

Reply:
Fixed by [b7f664a](https://github.com/wso2/carbon-apimgt/pull/11325/commits/b7f664af4e393bb1fe5d307dd9dc9c9107e372d6)